make/photon/db/docker-entrypoint.sh

```diff
@@ -89,26 +101,61 @@ EOSQL
         psql+=( --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" )
 
         echo
-        for f in /docker-entrypoint-initdb.d/*; do
-                case "$f" in
-                        *.sh)     echo "$0: running $f"; . "$f" ;;
-                        *.sql)    echo "$0: running $f"; "${psql[@]}" -f "$f"; echo ;;
-                        *.sql.gz) echo "$0: running $f"; gunzip -c "$f" | "${psql[@]}"; echo ;;
-                        *)        echo "$0: ignoring $f" ;;
-                esac
-                echo
-        done
+
+        if [ $2 == "true" ]; then
+                for f in /docker-entrypoint-initdb.d/*; do
+                        case "$f" in
+                                *.sh)     echo "$0: running $f"; . "$f" ;;
+                                *.sql)    echo "$0: running $f"; "${psql[@]}" -f "$f"; echo ;;
+                                *.sql.gz) echo "$0: running $f"; gunzip -c "$f" | "${psql[@]}"; echo ;;
+                                *)        echo "$0: ignoring $f" ;;
+                        esac
+                        echo
+                done
+        fi
 
         PGUSER="${PGUSER:-postgres}" \
-        pg_ctl -D "$PGDATA" -m fast -w stop
+        pg_ctl -D "$1" -m fast -w stop
 
         echo
         echo 'PostgreSQL init process complete; ready for start up.'
         echo
+
+}
+
+# look specifically for PG_VERSION, if exist and needs upgrade, use pg_upgrade to handle it. Otherwise just init db.
+if [ -s $PGDATA/PG_VERSION ]; then
+        PG_VERSION_OLD=$(cat $PGDATA/PG_VERSION);
+        if [ "`echo "${PG_VERSION_OLD} < $PG_VERSION_NEW" | bc`" -eq 1 ]; then
+                echo 'PostgreSQL start to move $PGDATA to $PGDATAOLD.'
+                mkdir -p $PGDATAOLD
+                chown -R postgres "$PGDATAOLD"
+                mv -f $PGDATA/* $PGDATAOLD
```

Comment:
What will happen using mv here when the upgrade process failed?

Reply:
how about cp to the `/tmp` directory, and let the migration happen in the `/tmp` directory, and after the migration is successful, replace the content in `PGDATA` with the migrated data?